cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java

```diff
@@ -0,0 +1,162 @@
+package com.amazonaws.services.crossregion.schemaregistry.kafkaconnect;
+
+import com.amazonaws.services.schemaregistry.common.AWSSerializerInput;
+import com.amazonaws.services.schemaregistry.common.Schema;
+import com.amazonaws.services.schemaregistry.common.configs.GlueSchemaRegistryConfiguration;
+import com.amazonaws.services.schemaregistry.deserializers.GlueSchemaRegistryDeserializationFacade;
+import com.amazonaws.services.schemaregistry.deserializers.GlueSchemaRegistryDeserializerImpl;
+import com.amazonaws.services.schemaregistry.exception.AWSSchemaRegistryException;
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializationFacade;
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializerImpl;
+import com.amazonaws.services.schemaregistry.utils.AWSSchemaRegistryConstants;
+
+import kotlinx.serialization.SerializationException;
+import lombok.Data;
+import lombok.Getter;
+
+import org.apache.kafka.connect.data.*;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.storage.Converter;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Data
+public class CrossRegionReplicationMM2Converter implements Converter {
+
+    private GlueSchemaRegistryDeserializationFacade deserializationFacade;
+    private GlueSchemaRegistrySerializationFacade serializationFacade;
+    private AwsCredentialsProvider credentialsProvider;
+    private GlueSchemaRegistryDeserializerImpl deserializer;
+    private GlueSchemaRegistrySerializerImpl serializer;
+    private boolean isKey;
+
+    /**
+     * Constructor used by Kafka Connect user.
+     */
+    public CrossRegionReplicationMM2Converter(){};
+
+    /**
+     * Constructor accepting AWSCredentialsProvider.
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    public CrossRegionReplicationMM2Converter(
```

Comment:
this constructor is only used for testing purpose right? lets convert it to package-private